xds/src/main/java/io/grpc/xds/LrsLoadBalancer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import android.annotation.SuppressLint;
+import io.grpc.ConnectivityState;
+import io.grpc.LoadBalancer;
+import io.grpc.Status;
+import io.grpc.util.ForwardingLoadBalancerHelper;
+import io.grpc.util.GracefulSwitchLoadBalancer;
+import io.grpc.xds.ClientLoadCounter.LoadRecordingSubchannelPicker;
+import io.grpc.xds.EnvoyProtoData.Locality;
+import io.grpc.xds.LrsLoadBalancerProvider.LrsConfig;
+import io.grpc.xds.XdsSubchannelPickers.ErrorPicker;
+import java.util.Objects;
+
+/**
+ * Load balancer for lrs policy.
+ */
+final class LrsLoadBalancer extends LoadBalancer {
+  private final LoadBalancer.Helper helper;
+  private GracefulSwitchLoadBalancer switchingLoadBalancer;
```

Comment:
`@Nullable`

Reply:
Ah. Strictly speaking all the following fields are nullable as they are only initiated after the first invocation of `handleResolvedAddresses(...)`. But in the context of our design and how LB policy instances are structured, we use `handleResolvedAddresses(...)` as the entry point instead of constructor (we cannot). `handleResolvedAddresses(...)` is always called immediately after an instance is created. This is unfortunate, feature development does not perfectly fit into our existing LoadBalancer APIs. I am feeling better to not have them being annotated with `@Nullable`.

Reply:
Then maybe `@CheckForNull` might be more suitable. After all,  we can not guarantee `handleResolvedAddresses` is always the first method to be called for the instance.

I don't care too much about the other fields, because they are only used in one method.